apps/src/utils.js

```diff
@@ -331,28 +331,38 @@ exports.browserSupportsCssMedia = function () {
  */
 exports.unescapeText = function(text) {
   var cleanedText = text;
-  cleanedText = cleanedText.replace(/<div>/gi, '\n'); // Divs generate newlines
-  cleanedText = cleanedText.replace(/<[^>]+>/gi, ''); // Strip all other tags
 
-  // This next step requires some explanation
+  // Handling of line breaks:
   // In multiline text it's possible for the first line to render wrapped or unwrapped.
   //     Line 1
   //     Line 2
-  //   Can render as either of:
+  //   Can render as any of:
   //     Line 1<div>Line 2</div>
+  //     Line 1<br><div>Line 2</div>
   //     <div>Line 1</div><div>Line 2</div>
   //
-  // But leading blank lines will always render wrapped and should be preserved
+  // Most blank lines are rendered as <div><br></div>
+  //     Line 1
   //
-  //     Line 2
   //     Line 3
-  //   Renders as
-  //    <div><br></div><div>Line 2</div><div>Line 3</div>
+  //   Can render as any of:
+  //     Line 1<div><br></div><div>Line 3</div>
+  //     Line 1<br><div><br></div><div>Line 3</div>
+  //     <div>Line 1</div><div><br></div><div>Line 3</div>
   //
-  // To handle this behavior we strip leading newlines UNLESS they are followed
-  // by another newline, using a negative lookahead (?!)
-  cleanedText = cleanedText.replace(/^\n(?!\n)/, ''); // Strip leading nondoubled newline
+  // Leading blank lines render wrapped or as placeholder breaks and should be preserved
+  //
+  //     Line 2
+  //   Renders as any of:
+  //    <br><div>Line 2</div>
+  //    <div><br></div><div>Line 2</div>
 
+  // First, convert every <div> tag that isn't at the very beginning of the string
+  // to a newline.  This avoids generating an incorrect blank line at the start
+  // if the first line is wrapped in a <div>.
+  cleanedText = cleanedText.replace(/(?!^)<div>/gi, '\n');
```

Comment:
Note: `(?!^)` is a regex negative assertion - in this case, it means "and we can't match the beginning of the text here."